### PR TITLE
MINI-3194 Automate analytics events

### PR DESCRIPTION
**Pull request by munir-rakuten**
# Description
Describe the changes in this pull request.
Created `CustomButtonViewWithArrow` as a common view in settings page and integrated RAT analytics on this section only as a demo
Screen Shot

## Links
MINI-3194

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by corycaywood — APPROVED** [review]
Looks good! 👍

**Review by Climbatize — APPROVED** [review]
It's a good concept, looks good to me

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/CustomButtonViewWithArrow.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package com.rakuten.tech.mobile.testapp.rat_wrapper
```
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/CustomButtonViewWithArrow.kt:1]: I think, we can move this package under `com.rakuten.tech.mobile.testapp.analytics` ?

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/CustomButtonViewWithArrow.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        val returnClick = super.performClick()
+        prepareEventForSend()
+        ratEvent?.let {
+            DemoAppAnalytics.init(AppSettings.instance.projectId).sendAnalytics(it)
+        }
+        return returnClick
+    }
+
+    override fun prepareEventForSend() {
+        if (ratEvent == null)
```
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/CustomButtonViewWithArrow.kt:54]: `ratEvent` is already non-nullable, so, we don't need to check it here.

**Review by khairul-alam-licon — APPROVED** [review]
Nice concept 👍

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/RATSwitch.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    override fun setCustomRatEvent(ratEvent: RATEvent) {
+        this.ratEvent = ratEvent
+    }
+
+    override fun onTouchEvent(event: MotionEvent?): Boolean {
+        when (event?.action) {
+            MotionEvent.ACTION_UP ->
+                /**
+                 * Check status
+                 */
```
[the remark this answers is not in the document]
Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/RATSwitch.kt]: removed it 🙂

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/CustomButtonViewWithArrow.kt (hunk 1 above)
Comment (munir-rakuten) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/rat_wrapper/CustomButtonViewWithArrow.kt]: Changed the directory 🙂

**Pull request merged**